server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -731,7 +759,7 @@ private void stopBeingLeader()
       lookupCoordinatorManager.stop();
       metadataRuleManager.stop();
       segmentsMetadataManager.stopPollingDatabasePeriodically();
-
+      exec.shutdownNow();
```

Comment:
This need not be shutdown here as each `DutiesRunnable` which runs on this executor individually checks whether it should REPEAT or STOP.
If we do want to shutdown to be sure, however, then we must also re-initialize it in `becomeLeader`.